cli/src/pcluster/validators/cluster_validators.py

```diff
@@ -1272,3 +1254,76 @@ def _validate(
             f"{compute_resource_name} must have the same inference accelerator manufacturer ({heterogeneous_values}).",
             failure_level=FailureLevel.ERROR,
         )
+
+
+class InstanceTypesListEFAValidator(Validator, _FlexibleInstanceTypesValidatorMixin):
+    """Validate EFA requirements for Flexible Instance Types."""
+
+    def _validate(
+        self,
+        compute_resource_name: str,
+        instance_types_info: Dict[str, InstanceTypeInfo],
+        efa_enabled: bool,
+    ):
+        """Check if EFA requirements are met.
+
+        Validation Failure is expected if EFA is ENABLED and at least one instance type defined in the compute resource
+        DOES NOT support EFA.
+        """
+        if efa_enabled:
+            all_instance_types = set(instance_types_info.keys())
+            instance_types_without_efa_support = {
+                instance_type_name
+                for instance_type_name, instance_type_info in instance_types_info.items()
+                if not instance_type_info.is_efa_supported()
+            }
+
+            # If all the instance types have EFA support, `instance_types_without_efa_support` should be empty
+            # --> No failure expected
+            # If all the instance types DO NOT have EFA support, `instance_types_without_efa_support` should be the same
```

Comment:
> If all the instance types DO NOT have EFA support, instance_types_without_efa_support should be the same as all_instance_types (Set difference of 0) --> No failure expected

Please fix this: this should be a failure because EFA is enabled on LT and NO instance will be launched, since none of them support EFA